independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/InterceptionProxyGenerator.java

```diff
@@ -371,7 +371,10 @@ private void createInterceptionSubclass(ClassOutput classOutput, InterceptionPro
 
                 BytecodeCreator notConstructed = mc.ifFalse(
                         mc.readInstanceField(constructedField.getFieldDescriptor(), mc.getThis())).trueBranch();
-                notConstructed.returnValue(notConstructed.invokeSpecialMethod(method, notConstructed.getThis(), args));
+                ResultHandle notConstructedResult = method.declaringClass().isInterface()
+                        ? notConstructed.invokeInterfaceMethod(method, notConstructed.getThis(), args)
```

Comment:
Maybe this should be `invokeSpecialInterfaceMethod()`? I"m not exactly sure what this code does (which sounds weird, because I wrote it :-) ), I'll take a look again tomorrow.

Reply:
My understanding of the problem is that `invokespecial` may not be used in this specific case.

Reply:
Yeah, so I think what we should do here is the same as in client proxies: if the method to which we're supposed to delegate is `abstract`, we should just throw an exception. Delegation is not gonna succeed anyway, there's no method body to invoke. (Isn't it funny how we're converging to the correct behavior of client proxies here? I thought I'm smart. I'm clearly not :-) )